GeeksCoreLibrary/Modules/MeasurementProtocol/Services/MeasurementProtocolService.cs

```diff
@@ -0,0 +1,212 @@
+
+            if (dataTable.Rows.Count == 0)
+            {
+                return;
+            }
+
+            var replaceData = new Dictionary<string, object>()
+            {
+                {"payment_method", dataTable.Rows[0].Field<string>("title") }
+            };
+
+            await SendMeasurement(orderProcessSettings, orderProcessSettings.MeasurementProtocolAddPaymentInfoJson, orderProcessSettings.MeasurementProtocolItemJson, replaceData, shoppingBasket, shoppingBasketLines, shoppingBasketSettings);
+        }
+
+        /// <inheritdoc />
+        public async Task PurchaseEventAsync(OrderProcessSettingsModel orderProcessSettings, WiserItemModel shoppingBasket, List<WiserItemModel> shoppingBasketLines, ShoppingBasketCmsSettingsModel shoppingBasketSettings, string transactionId)
+        {
+            var tax = await shoppingBasketsService.GetPriceAsync(shoppingBasket, shoppingBasketLines, shoppingBasketSettings, ShoppingBasket.PriceTypes.VatOnly);
+
+            var replaceData = new Dictionary<string, object>()
+            {
+                {"tax_price", tax },
+                {"transaction_id", transactionId }
+            };
+
+            await SendMeasurement(orderProcessSettings, orderProcessSettings.MeasurementProtocolPurchaseJson, orderProcessSettings.MeasurementProtocolItemJson, replaceData, shoppingBasket, shoppingBasketLines, shoppingBasketSettings);
+        }
+
+        /// <summary>
+        /// Build the request and send it to Google Analytics.
+        /// </summary>
+        /// <param name="orderProcessSettings">The settings of the orde process.</param>
+        /// <param name="eventJson">The Json to use for the event.</param>
+        /// <param name="itemJson">The Json to use for each item.</param>
+        /// <param name="replaceData">The event specific data that can be used for replacements.</param>
+        /// <param name="shoppingBasket">The shopping basket to use for the event.</param>
+        /// <param name="shoppingBasketLines">The shopping basket lines to use for the event.</param>
+        /// <param name="shoppingBasketSettings">The settings of the shopping basket.</param>
+        /// <returns></returns>
+        private async Task SendMeasurement(OrderProcessSettingsModel orderProcessSettings, string eventJson, string itemJson, Dictionary<string, object> replaceData, WiserItemModel shoppingBasket, List<WiserItemModel> shoppingBasketLines, ShoppingBasketCmsSettingsModel shoppingBasketSettings)
+        {
+            if (String.IsNullOrWhiteSpace(orderProcessSettings.MeasurementProtocolMeasurementId) || String.IsNullOrWhiteSpace(orderProcessSettings.MeasurementProtocolApiSecret) || String.IsNullOrWhiteSpace(eventJson) || String.IsNullOrWhiteSpace(itemJson))
+            {
+                return;
+            }
+
+            var clientId = shoppingBasket.GetDetailValue("google-cid");
```

Comment:
Kun je een constante maken van `google-cid` en op alle plekken in de GCL kijken waar dit gebruikt wordt en dan daar die constante gebruiken?